### ISSUE TITLE
Quadrigacx

### ISSUE THREAD

**Issue by marinsokol5** [kroitor/ccxt#32]
The same thing is happening with quadrigacx and its private api
```File "C:\Program Files (x86)\Python36-32\lib\site-packages\ccxt\__init__.py", line 8766, in request
    signature = self.hmac (self.encode (request), self.secret)
  File "C:\Program Files (x86)\Python36-32\lib\site-packages\ccxt\__init__.py", line 389, in hmac
    h = hmac.new (secret, request, algorithm)
  File "C:\Program Files (x86)\Python36-32\lib\hmac.py", line 144, in new
    return HMAC(key, msg, digestmod)
  File "C:\Program Files (x86)\Python36-32\lib\hmac.py", line 42, in __init__
    raise TypeError("key: expected bytes or bytearray, but got %r" % type(key).__name__)
TypeError: key: expected bytes or bytearray, but got 'str'
```

And just in case you didnt see it I'l post it again.
I have a problem with signature in cex.
Not sure from which side is it.Mine or yours.
Im testing for method:  `get_myfee/ `  in cex
This is what i get as a response
`{'e': 'get_myfee', 'error': 'Invalid signature', 'ok': 'error'}`
Api key and secret should be correctly inputed.I have checked.Might be a problem with computing the signature?

**Comment by kroitor** [comment]
And just in case you didnt see it I'l post it again.
> I have a problem with signature in cex.
> Not sure from which side is it.Mine or yours.
> Im testing for method: get_myfee/ in cex
> This is what i get as a response
> {'e': 'get_myfee', 'error': 'Invalid signature', 'ok': 'error'}
> Api key and secret should be correctly inputed.I have checked.Might be a problem with computing the signature?

I replied to your question about CEX there on your issue https://github.com/kroitor/ccxt/issues/31
I'll duplicate that here:
You probably forgot to set the `uid`. CEX.IO requires that. See [API Keys Setup](https://github.com/kroitor/ccxt/wiki/Manual#api-keys-setup) section in the manual.

To test if this is the case with uid, try the following code:
```Python
import ccxt
cex = ccxt.cex ({
    'apiKey': 'eqCv267WySlu577JnFbGK2RQzIs', # this is standard
    'secret': 'pZnbuNEm5eE4W1VRuFQvZEiFCA', # this is also standard
    'uid': 'up105393824', # for CEX.IO you should set uid as string
	
})
cex.fetch_balance ()
```
> ```{'timestamp': '1500262307', 'username': 'up105393824', 'BTC': {'available': '0.00000000', 'orders': '0.00000000'}, 'USD': {'available': '0.00', 'orders': '0.00'}, 'EUR': {'available': '0.00', 'orders': '0.00'}, 'GBP': {'available': '0.00', 'orders': '0.00'}, 'GHS': {'available': '0.00000000', 'orders': '0.00000000'}, 'RUB': {'available': '0.00', 'orders': '0.00'}, 'ETH': {'available': '0.00000000', 'orders': '0.00000000'}}```

Please, test the code above, then try your credentials and report if everything is ok or not. Thank you very much! Appreciate your help!

As for QuadrigaCX, fixing it right now, should be ok very soon.

**Comment by kroitor** [comment]
QuadrigaCX should work now, code for testing:
```
import ccxt
quadrigacx = ccxt.quadrigacx ({
    'apiKey': 'jKvWkMqrOj', # this is standard
    'secret': 'f65a2e3bf3c73171ee14e389314b2f78', # this is also standard
    'uid': '395037', # you should set uid as string	
})
quadrigacx.fetch_balance ()
```
> ```{'btc_available': '0.00000000', 'btc_reserved': '0.00000000', 'btc_balance': '0.00000000', 'etc_available': '0.00000000', 'etc_reserved': '0.00000000', 'etc_balance': '0.00000000', 'eth_available': '0.00000000', 'eth_reserved': '0.00000000', 'eth_balance': '0.00000000', 'cad_available': '0.00', 'cad_reserved': '0.00', 'cad_balance': '0.00', 'usd_available': '0.00', 'usd_reserved': '0.00', 'usd_balance': '0.00', 'xau_available': '0.000000', 'xau_reserved': '0.000000', 'xau_balance': '0.000000', 'fee': '0.5000', 'fees': {'btc_cad': '0.5000', 'btc_usd': '0.5000', 'eth_btc': '0.2000', 'eth_cad': '0.5000'}}```

Please report if everything is ok.
